javalib/src/main/scala/java/lang/process/UnixProcessFactory.scala

```diff
@@ -250,14 +250,26 @@ private[process] object UnixProcessFactory {
     }
   }
 
-  private def nullTerminate(
+  /* unix-like operating systems define argv and envp as arrays of
+   * pointers to char, where the final element is a null pointer.
+   *
+   * This method converts a List[String] representing elements for either
+   * argv or envp to a form which can be passed directly to posix_spawn or
+   * one of the exec* C functions. That is, a C pointer_to/array_of
+   * pointers to char.
+   * 
+   * Most noticeably, it _does_not_ return a Scala Native CArray[T, N <: Nat].
+   */
+
+  private def toUnixArgvLike(
       list: java.util.List[String]
   )(implicit z: Zone) = {
     val res: Ptr[CString] = alloc[CString]((list.size() + 1))
     val li = list.listIterator()
-    while (li.hasNext()) {
+    while (li.hasNext())
       !(res + li.nextIndex()) = toCString(li.next())
-    }
+
+    !(res + list.size()) = null // NULL ptr, not NUL character (ASCII 0)
```

Comment:
questions:
- did it work before? did `alloc` null-initialize memory even though it doesn't promise to?
- what's the difference between null and NUL in this context?

Reply:
! what's the difference between null and NUL in this context?

Trying to head off the usual "What is NULL/null/NUL" discussion/confusion.  
Agreed Scala null ought to be pretty clear but there is always someone.

I weaseled and used Scala null rather than `clib#stddef` NULL with an `asInstanceOf[Ptr[CChar]`
to save a few instructions. Compiler let me get away with that.

Reply:
Good sleuthing!     Thank you & Ekrich.

I'll move this to Draft and re-work it later, using the `update()` suggestion.


@kitbellew   do you remember the context of our week or so discussion of `alloc` clearing?
Is there something there which could be simplified based on the fact that we have re-established
that `alloc` does clear?    A lot of water under the bridge since then.

It may have been around another discussion of 'update'.

Reply:
There is some satisfaction to knowing that the documentation is correct and that the World did
not change and break in the night.